api/pro_api.go

```diff
@@ -64,6 +66,33 @@ func (c *proClient) CreateUser(ctx context.Context) (*protos.UserDataResponse, e
 		return nil, err
 	}
 	return resp, nil
+}
+
+// UserData is used to get user data
+// this will be also save data to user config
+func (c *proClient) UserData(ctx context.Context) (*protos.UserDataResponse, error) {
+	var resp *protos.UserDataResponse
+	err := c.Get(ctx, "/user-data", nil, &resp)
+	if err != nil {
+		slog.Error("Error in UserData: %v", "err", err)
+		return nil, err
```

Comment:
We should add context that is more specific to the error. Using the `%w` flag in `fmt.Errorf` will wrap the error and include the stack trace, which makes debugging easier.
```suggestion
		err = fmt.Errorf("sending user data request: %w", err)
		slog.Error(err)
		return nil, err
```

Reply:
I don't think we can do something like this.
`slog.Error(err)`

For now I just used 
`slog.Error("seding user data request: %v", "err", err)`